python/neml2/pyzag/interface.py

```diff
@@ -0,0 +1,287 @@
+from pyzag import nonlinear
+
+import neml2
+from neml2.tensors import Tensor
+from neml2.tensors import LabeledAxisAccessor as AA
+
+import torch
+
+
+def assemble_vector(axis, tensors):
```

Comment:
```suggestion
def assemble_vector(axis, tensors, warn_unused=True):
```
I suggest adding an error check. It's up to you to make it a warning or error, but I think it's important to be there. There are two cases:
1. The `axis` has a variable that isn't provided by `tensors`.
2. The `tensors` provides a variable that isn't defined on `axis`.

The actual behavior is debatable, but I guess case 1 is fine -- just fill zeros by default, and case 2 might indicate a user error.